### PR TITLE
skip redownload if file exists

### PR DESCRIPTION
**Pull request by dfang**
skip redownload if file exists for go, bun, deno, java
for node and python, they are controlled by NODE_BUILD_BUILD_PATH and PYTHON_BUILD_BUILD_PATH

### CODE REVIEW

File: src/plugins/core/java.rs [hunk 1]
```diff
@@ -110,6 +110,15 @@ impl JavaPlugin {
         let filename = m.url.split('/').last().unwrap();
         let tarball_path = tv.download_path().join(filename);
 
+        if tarball_path.exists() {
+            pr.set_message(format!(
+                "file {} exists, skipping download",
+                tarball_path.to_string_lossy()
+            ));
+            hash::ensure_checksum_sha256(&tarball_path, &m.sha256)?;
```
Comment (jdx) [inline src/plugins/core/java.rs]: we need to still check the checksum to see if it is valid, if not we should redownload it. For languages that do not support checksums we should always redownload.

**Review by jdxcode — CHANGES_REQUESTED** [review]

File: src/plugins/core/java.rs (hunk 1 above)
Comment (dfang) [reply src/plugins/core/java.rs]: currently go and java plugin have checksum support, deno and bun are downloading from GitHub,  bun has SHASUMS256.txt in release assets, but deno doesn't

Comment (dfang) [reply src/plugins/core/java.rs]: If the network is not smooth, re-downloading might be very slow. I will try to add checksum support or env like RTX_GO_SKIP_CHECKSUM

File: src/tool.rs [hunk 2]
```diff
@@ -333,7 +333,7 @@ impl Tool {
 
     fn create_install_dirs(&self, tv: &ToolVersion) -> Result<()> {
         let _ = remove_all_with_warning(tv.install_path());
-        let _ = remove_all_with_warning(tv.download_path());
+        // let _ = remove_all_with_warning(tv.download_path());
```
Comment (dfang) [inline src/tool.rs]: this applies to all plugins? not only core plugins?

**Pull request closed without merging**